client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -123,6 +125,17 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow, stepName } ) => {
 			}
 		}
 
+		if ( isDesignFirstFlow ) {
+			allDesigns.designs = allDesigns.designs.filter( ( design ) => design.is_premium === false );
+
+			allDesigns.designs = allDesigns.designs.map( ( design ) => {
+				design.style_variations = design.style_variations.filter(
```

Comment:
IDE warning: "Object is possibly undefined." -- I think this might fix it:
```suggestion
				design.style_variations = design.style_variations?.filter(
```